builder/builder/builder.go

```diff
@@ -359,14 +362,21 @@ func (b *Builder) subscribeToRelayForConstraints(relayBaseEndpoint string) error
 		}
 
 		for _, constraint := range constraintsSigned {
+			// Check if the signature is valid against any of the authorized pubkeys
+			for _, pubkey := range b.slotConstraintsPubkeys {
+				valid, err := constraint.VerifySignature(pubkey)
+				if err != nil || !valid {
+					log.Error("Failed to verify constraint signature", "err", err)
+					continue
+				}
```

Comment:
Why do we just continue in this case?

Reply:
You're right, we should not only move on to the next pubkey but if there is no valid signature, the constraint should be discarded. Addressed in latest commit